### PR TITLE
Consider my suggestions #1.

### PR DESCRIPTION
**Pull request by revug**

### CODE REVIEW

**Comment by revug** [comment]
Also you can edit:
SC0027::Send `;    ; You don't need 'Up' here.

File: dialog.ahk [hunk 1]
```diff
@@ -10,19 +10,40 @@ Gui, Show, CenterAutosize, Update
...
+		sleep 100
+	if request.readyState != 4 {
+		MsgBox, Connection to the server failed, try to update later.
+		if request.readyState != 4  ; We could  get the response while user was reading msgbox's text.
+			ExitApp
+		else MsgBox, A miracle has just happened, we've got the update!
+	}
+  	text := request.ResponseText
+	;-------------------------------------------------------------
+	; What were you doing here? You've just made "urrentVersion = 1.6" instead of "currentVersion = 1.6"
```
Comment (iQuerz) [inline dialog.ahk:32]: First of all, thank you very much. Everything you said here is very helpful and I am really thankful. Hey, nobody knows everything on this world. 
BUT - I wanted to add that I had a really weird issue with WinHttpRequest, as it was returning the whole text with a "?" character in front of the whole file. That's the reason I added the (maybe silly) first character removal part. 
Also, keep in mind that I'm relatively new to AHK and I didn't have anyone else to help me test my code so everything that I pushed on here worked on my PC. That's a requirement before commiting.

Comment (revug) [reply dialog.ahk:32]: > BUT - I wanted to add that I had a really weird issue with WinHttpRequest, as it was returning the whole text with a "?" character in front of the whole file. That's the reason I added the (maybe silly) first character removal part. 

Ok, I understand now.

> Also, keep in mind that I'm relatively new to AHK and I didn't have anyone else to help me test my code...

Don't worry, I'm not a programmer at all, just use a little bit Autohotkey for my needs. I told you about things that mostly present in the documentation that I hope you'll learn soon. For example:
https://www.autohotkey.com/docs/commands/URLDownloadToFile.htm#XHR
https://www.autohotkey.com/docs/commands/FileAppend.htm#Remarks
https://www.autohotkey.com/docs/commands/Gui.htm#GuiClose
https://www.autohotkey.com/docs/Hotkeys.htm#combo

> ...so everything that I pushed on here worked on my PC. That's a requirement before commiting.

Didin't catch. You haven't tested these suggestions yet? I've tested.

**Comment by iQuerz** [comment]
I've tested, reviewed and merged your suggestions. I send you a big thank you for steering me in a useful direction.  
Next up will be cleaning up the comments and other parts of the code using your tips.  
I am aware that I didn't read the documentation fully, and I could spend a bit more time on it.

**Pull request merged**